program/src/instruction.rs

```diff
@@ -237,6 +237,31 @@ pub enum MangoInstruction {
         reduce_only: bool,
     },
 
+    /// Place an order on a perp market Improved over PlacePerpOrder
+    /// by reducing the tx size, and using max base/quote.
+    ///
+    /// For OrderTypes Limit, PostOnly, PostOnlySlide the
+    /// `max_base_quantity` will simply be `quantity`, and the
+    /// `limit_price` be `price`.
+    ///
+    /// The max parameters are useful when using instantaneous
+    /// OrderTypes (IoC or Market).
+    /// Goal is to allow orders with thresholds, without knowing
+    /// the quantity nor price in advance.
+    /// It will check wether it fits within the provided `limit_price`,
+    /// It will optionally checks wether it's fully filled for IoC.
+    PlacePerpOrderV2 {
```

Comment:
note: I'll also make a PlacePerpOrderV2 for the time-in-force feature. Let's get both new features into that new instruction.